### PR TITLE
Documentation for new agent upgrade module and task manager module

### PR DESCRIPTION
**Pull request by TomasTurina**
## Description

Changes included:

- [x] Add configuration page of the new upgrade module.

- [x] Add configuration page of the new tasks module.

- [x] List new modules in the `wazuh-modules` section.

- [x] Add new tasks table in the `wazuh-db` section.

- [x] Add note of cluster compatibility since v4.1.0.

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [x] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

## Note to the reviewer

This PR includes changes to the `redirect.js` script that need to be included in all production branches.

### CODE REVIEW

File: source/user-manual/reference/ossec-conf/task-manager.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+============
+
+.. topic:: XML section name
+
+	.. code-block:: xml
+
+		<task-manager>
+		</task-manager>
+
+The task manager module is responsible for managing all the tasks that run on the agents. This module is only available on the manager side, on a stand-alone manager or on a master node in a cluster configuration. This module will be enabled always and cannot be deactivated. This section does not need to be defined but needs to be defined in order to change the default values.
```
Comment (nicolaspapp) [inline source/user-manual/reference/ossec-conf/task-manager.rst:15]: Last sentence. `This section only needs to be defined in order to change the default values`

**Review by nicolaspapp — CHANGES_REQUESTED** [review]

File: source/user-manual/agents/remote-upgrading/agent-upgrade-module.rst [hunk 2]
```diff
@@ -0,0 +1,183 @@
+.. Copyright (C) 2020 Wazuh, Inc.
```
[the remark this answers is not in the document]
Comment (davidjiglesias) [reply source/user-manual/agents/remote-upgrading/agent-upgrade-module.rst:1]: I think the suggestion you made is good enough and should not be a problem! I think it is important to explain to the user what is the purpose of the module and how to activate/deactivate it in the agent side.

File: source/user-manual/reference/ossec-conf/agent-upgrade.rst [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+=============
+
+.. topic:: XML section name
+
+	.. code-block:: xml
+
+		<agent-upgrade>
+		</agent-upgrade>
+
+The agent upgrade module is responsible for carrying out the entire agent upgrade process remotely, both downloading and/or sending the WPK from the manager to the agent as well as sending the response with the result of an upgrade from the agent to the manager. In the manager side, this module will be enabled always and cannot be deactivated. This section does not need to be defined but needs to be defined in order to change the default values.
```
[the remark this answers is not in the document]
Comment (TomasTurina) [reply source/user-manual/reference/ossec-conf/agent-upgrade.rst]: Done! I've changed this as you suggested.

File: source/user-manual/reference/ossec-conf/task-manager.rst (hunk 1 above)
Comment (TomasTurina) [reply source/user-manual/reference/ossec-conf/task-manager.rst]: Changed!

**Review by nicolaspapp — APPROVED** [review]

**Review by davidfmiranda — APPROVED** [review]
LGTM

**Pull request merged**